src/julia_gc.c

```diff
@@ -387,6 +388,36 @@ void MarkJuliaObj(void * obj)
 // from which freed objects are then marked. Hence, we add additional checks
 // when traversing GAP master pointer and bag objects that this happens
 // only for live objects.
+//
+// All stacks in Julia are associated with tasks and we can use
+// jl_task_stack_buffer() to retrieve the buffer information for that stack.
+// That said, in a couple of cases these need adjustments.
+//
+// 1. The stack buffer of the root task of the main thread, when started from
+//    GAP can extend past the point where Julia believes it ends. Therefore,
+//    for that stack, we use GapBottomStack instead.
+// 2. If it is the current task of the current thread, we know where exactly
+//    the lowest valid address of the stack is and do not need to scan the
+//    entire stack buffer.
+//
+// In addition, stack buffers can include guard pages, which trigger a
+// segmentation fault when accessed. We intercept such segmentation faults
+// and terminate stack scanning early for those.
+//
+// In GAP terminology, we use "bottom" to describe the end of the stack
+// where it starts and "top" to describe the current stack pointer. This
+// means that the "bottom" of the stack is the highest address on most
+// modern architectures and "top" is the lowest. When we treat the stack
+// as an abstract buffer, "start" and "end" refer to its lowest and
+// highest address, respectively.
+//
+//   +------------------------------------------------+
+//   | guard |  unused area            | active stack |
+//   | pages |                         | frames       |
```

Comment:
How about adding in the growth direction?
```suggestion
//   | pages |     <--- growth ---     | frames       |
```

Reply:
Good idea.